pkg/validation/cleanuppolicy/validate.go

```diff
@@ -0,0 +1,76 @@
+package cleanuppolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	kyvernov1alpha1 "github.com/kyverno/kyverno/api/kyverno/v1alpha1"
+	"github.com/kyverno/kyverno/pkg/auth"
+	"github.com/kyverno/kyverno/pkg/clients/dclient"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/discovery"
+)
+
+// FetchClusteredResources retieves the list of clustered resources
+func FetchClusteredResources(logger logr.Logger, client dclient.Interface) (sets.String, error) {
+	res, err := discovery.ServerPreferredResources(client.Discovery().DiscoveryInterface())
+	if err != nil {
+		if discovery.IsGroupDiscoveryFailedError(err) {
+			err := err.(*discovery.ErrGroupDiscoveryFailed)
+			for gv, err := range err.Groups {
+				logger.Error(err, "failed to list api resources", "group", gv)
+			}
+		} else {
+			return nil, err
+		}
+	}
+	clusterResources := sets.NewString()
+	for _, resList := range res {
+		for _, r := range resList.APIResources {
+			if !r.Namespaced {
+				clusterResources.Insert(r.Kind)
+			}
+		}
+	}
+	return clusterResources, nil
+}
+
+// Validate checks policy is valid
+func Validate(ctx context.Context, logger logr.Logger, client dclient.Interface, policy kyvernov1alpha1.CleanupPolicyInterface) error {
+	clusteredResources, err := FetchClusteredResources(logger, client)
+	if err != nil {
+		return err
+	}
+	if err := ValidatePolicy(clusteredResources, policy); err != nil {
+		return err
+	}
+	if err := ValidateAuth(ctx, client, policy); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidatePolicy checks the policy and rules declarations for required configurations
+func ValidatePolicy(clusterResources sets.String, policy kyvernov1alpha1.CleanupPolicyInterface) error {
```

Comment:
Do we need to expose this and `ValidateAuth`?

Reply:
Dunno, it doesn't hurt I guess ?

Reply:
True it wouldn't impact anything. Would be good to create private functions if they are not used externally IMHO:)

Reply:
I'll fix this

Reply:
Fixed.